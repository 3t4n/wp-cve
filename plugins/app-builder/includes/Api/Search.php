<?php


/**
 * class Search
 *
 * @link       https://appcheap.io
 * @since      1.0.0
 *
 * @author     AppCheap <ngocdt@rnlab.io>
 */

namespace AppBuilder\Api;

defined( 'ABSPATH' ) || exit;

class Search extends Base {
	public function __construct() {
		$this->namespace = APP_BUILDER_REST_BASE . '/v1';
	}

	/**
	 * Registers a REST API route
	 *
	 * @since 1.0.0
	 */
	public function register_routes() {

		/**
		 * search
		 *
		 * @author Ngoc Dang
		 * @since 1.0.0
		 */
		register_rest_route( $this->namespace, 'search', array(
			'methods'             => \WP_REST_Server::READABLE,
			'callback'            => array( $this, 'search' ),
			'permission_callback' => '__return_true',
		) );

		/**
		 * Post search
		 *
		 * @author Ngoc Dang
		 * @since 1.0.0
		 */
		register_rest_route( $this->namespace, 'post-search', array(
			'methods'             => \WP_REST_Server::READABLE,
			'callback'            => array( $this, 'post_search' ),
			'permission_callback' => '__return_true',
		) );

		/**
		 * Term search
		 *
		 * @author Ngoc Dang
		 * @since 1.0.0
		 */
		register_rest_route( $this->namespace, 'term-search', array(
			'methods'             => \WP_REST_Server::READABLE,
			'callback'            => array( $this, 'term_search' ),
			'permission_callback' => '__return_true',
		) );
	}

	/**
	 * @param $request
	 *
	 * @return array
	 * @since 1.0.0
	 * @author Ngoc Dang
	 */
	public function search( $request ): array {
		$posts = $this->post_search( $request );
		$terms = $this->term_search( $request );

		$results = [];

		array_push( $results, ...$posts );

		if ( ! is_wp_error( $terms ) ) {
			array_push( $results, ...$terms );
		}

		return $results;
	}

	/**
	 * @param $request
	 *
	 * @return array
	 * @since 1.0.0
	 * @author Ngoc Dang
	 */
	public function post_search( $request ): array {

		// Post
		$search    = $request->get_param( 'search' ) ? $request->get_param( 'search' ) : '';
		$limit     = $request->get_param( 'limit' ) ? (int) $request->get_param( 'limit' ) : 10;
		$post_type = is_array( $request->get_param( 'post_type' ) ) ? $request->get_param( 'post_type' ) : array();

		if ( count( $post_type ) == 0 ) {
			return array();
		}

		$return = array();
		$args   = array(
			's'              => $search,
			'posts_per_page' => $limit,
			'post_status'    => "publish",
			'post_type'      => $post_type
		);
		$query  = new \WP_Query( $args );

		$posts = $query->get_posts();

		foreach ( $posts as $post ) {
			$newPost = array();

			$newPost['id']    = $post->ID;
			$newPost['title'] = $post->post_title;
			$newPost['type']  = $post->post_type;

			$return[] = $newPost;
		}

		return $return;
	}

	/**
	 *
	 * Search Terms
	 *
	 * @param $request
	 *
	 * @return int[]|string|string[]|WP_Error|WP_Term[]
	 * @author Ngoc Dang
	 * @since 1.0.0
	 */
	public function term_search( $request ) {

		$search   = $request->get_param( 'search' ) ? $request->get_param( 'search' ) : '';
		$taxonomy = is_array( $request->get_param( 'taxonomy' ) ) ? $request->get_param( 'taxonomy' ) : array();

		if ( count( $taxonomy ) == 0 ) {
			return array();
		}

		$terms = get_terms( array(
			'taxonomy'   => $taxonomy,
			'hide_empty' => false,
			'search'     => $search
		) );

		$return = array();

		foreach ( $terms as $term ) {
			$newTerm = array();

			$newTerm['id']    = $term->term_id;
			$newTerm['title'] = $term->name;
			$newTerm['type']  = $term->taxonomy;

			$return[] = $newTerm;
		}

		return $return;
	}
}
