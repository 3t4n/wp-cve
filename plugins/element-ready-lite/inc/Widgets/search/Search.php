<?php

namespace Element_Ready\Widgets\search;

use Elementor\Widget_Base;
use Elementor\Controls_Manager;
use Elementor\Group_Control_Typography;
use Elementor\Group_Control_Border;

if (!defined('ABSPATH')) exit;

class Search extends Widget_Base
{


    public $base;

    public function get_name()
    {
        return 'element-ready-popup-search';
    }
    public function get_keywords()
    {
        return ['element ready', 'popup search', 'search', 'popup'];
    }
    public function get_title()
    {
        return esc_html__('ER PopUp Search', 'element-ready-lite');
    }

    public function get_icon()
    {
        return 'eicon-search';
    }

    public function get_script_depends()
    {
        return [
            'element-ready-core',
        ];
    }

    public function get_style_depends()
    {

        wp_register_style('eready-search-pop', ELEMENT_READY_ROOT_CSS . 'widgets/search.css');
        return ['eready-search-pop'];
    }

    public function get_categories()
    {
        return ['element-ready-addons'];
    }
    public function layout()
    {
        return [

            'style1'   => esc_html__('style1', 'element-ready-lite'),

        ];
    }

    protected function register_controls()
    {

        $this->start_controls_section(
            'menu_layout',
            [
                'label' => esc_html__('Layout', 'element-ready-lite'),
            ]
        );


        $this->add_control(
            '_style',
            [
                'label' => esc_html__('Style', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::SELECT,
                'default' => 'style1',
                'options' => $this->layout()
            ]
        );

        $this->end_controls_section();


        $this->start_controls_section(
            'header_search_section',
            [
                'label' => esc_html__('Header Search', 'element-ready-lite'),

            ]
        );


        $this->add_control(
            'custom_search_templte',
            [
                'label' => esc_html__('Custom template', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::SWITCHER,
                'label_on' => esc_html__('Show', 'element-ready-lite'),
                'label_off' => esc_html__('Hide', 'element-ready-lite'),
                'return_value' => 'yes',
                'default' => '',
            ]
        );

        $this->add_control(
            'popup_search_template_id',
            [
                'label'     => esc_html__('Select Content Template', 'element-ready-lite'),
                'type'      => Controls_Manager::SELECT,
                'default'   => '0',
                'options'   => element_ready_elementor_template(),
                'condition' => [
                    'custom_search_templte' => ['yes']
                ],
                'description' => esc_html__('Please select elementor templete from here, if not create elementor template from menu', 'element-ready-lite')

            ]
        );

        $this->add_control(
            'search_display',
            [
                'label'     => esc_html__('Device Breakpoint', 'element-ready-lite'),
                'type'      => Controls_Manager::SELECT,
                'default'   => '',
                'options'   => [
                    '' => esc_html__('No Action', 'element-ready-lite'),
                    'd-xl-none' => esc_html__('Extra Large', 'element-ready-lite'),
                    'd-lg-none' => esc_html__('Large', 'element-ready-lite'),
                    'd-md-none' => esc_html__('Medium', 'element-ready-lite'),
                    'd-sm-none' => esc_html__('Small', 'element-ready-lite'),

                ],

            ]
        );

        $this->add_control(
            'header_search_text',
            [

                'label'       => esc_html__('Text', 'element-ready-lite'),
                'type'        => \Elementor\Controls_Manager::TEXT,
                'placeholder' => esc_html__('Search', 'element-ready-lite'),
                'default'     => esc_html__('Search', 'element-ready-lite')

            ]
        );

        $this->add_control(
            'header_search_close_text',
            [

                'label'       => esc_html__('Close Text', 'element-ready-lite'),
                'type'        => \Elementor\Controls_Manager::TEXT,
                'placeholder' => esc_html__('Close', 'element-ready-lite'),
                'default'     => esc_html__('close', 'element-ready-lite')

            ]
        );

        $this->add_control(
            'header_search_icon',
            [
                'label' => esc_html__('Icon', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::ICONS,

            ]
        );

        $this->add_control(
            'header_close_icon',
            [
                'label' => esc_html__('Close Icon', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::ICONS,

            ]
        );

        $this->add_control(
            'header_search_logo',
            [
                'label' => esc_html__('Choose logo', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::MEDIA,
            ]
        );

        $this->end_controls_section();

        $this->start_controls_section(
            'menu_search_container_section',
            [
                'label' => esc_html__('Popup Container', 'element-ready-lite'),
                'tab'   => Controls_Manager::TAB_STYLE,

            ]
        );

        $this->add_control(
            '__search_menu_container_po_bgcolor_',
            [

                'label'         => esc_html__('Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box' => 'background-color: {{VALUE}};',

                ],
            ]
        );

        $this->add_group_control(
            \Elementor\Group_Control_Box_Shadow::get_type(),
            [
                'name' => '__search_menu_container_po_n_box_shadow',
                'label' => esc_html__('Box Shadow', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-box',
            ]
        );

        $this->end_controls_section();

        $this->start_controls_section(
            'menu_search_popup_icons_section',
            [
                'label' => esc_html__('Search interface', 'element-ready-lite'),
                'tab'   => Controls_Manager::TAB_STYLE,

            ]
        );


        $this->add_control(
            '__search_menu_icon_color_',
            [

                'label'         => esc_html__('Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-open i' => 'color: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-open span' => 'color: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_menu_icon_hovern_color_',
            [

                'label'         => esc_html__('Hover Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [
                    '{{WRAPPER}} .element-ready-search-open:hover i' => 'color: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-open:hover span' => 'color: {{VALUE}};',
                ],
            ]
        );

        $this->add_control(
            '__search_menu_i_icon_bgcolor_',
            [

                'label'         => esc_html__('Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-open' => 'background: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_menu_i_hover_icon_bgcolor_',
            [

                'label'         => esc_html__('Hover Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-open:hover' => 'background: {{VALUE}};',

                ],
            ]
        );

        $this->add_responsive_control(
            '__search_menu_icon__border_radius',
            [
                'label'     => esc_html__('Border Radius', 'element-ready-lite'),
                'type'      => Controls_Manager::DIMENSIONS,
                'size_units' => ['px', '%', 'em'],
                'selectors' => [
                    '{{WRAPPER}} .element-ready-search-open' => 'border-radius: {{TOP}}{{UNIT}} {{RIGHT}}{{UNIT}} {{BOTTOM}}{{UNIT}} {{LEFT}}{{UNIT}};',

                ],
                'separator' => 'before',
            ]
        );

        $this->add_group_control(
            \Elementor\Group_Control_Border::get_type(),
            [
                'name' => '__search_menu_icon__section_border',
                'label' => esc_html__('Border', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-open',
            ]
        );

        $this->add_responsive_control(
            '__search_menu_iccon_section_margin',
            [
                'label'      => esc_html__('Margin', 'element-ready-lite'),
                'type'       => Controls_Manager::DIMENSIONS,
                'size_units' => ['px', '%', 'em'],
                'selectors'  => [
                    '{{WRAPPER}} .element-ready-search-open' => 'margin: {{TOP}}{{UNIT}} {{RIGHT}}{{UNIT}} {{BOTTOM}}{{UNIT}} {{LEFT}}{{UNIT}};',

                ],
                'separator' => 'before',
            ]
        );

        $this->add_responsive_control(
            '__search_menu_icon_section_padding',
            [
                'label'      => esc_html__('Padding', 'element-ready-lite'),
                'type'       => Controls_Manager::DIMENSIONS,
                'size_units' => ['px', '%', 'em'],
                'selectors'  => [

                    '{{WRAPPER}} .element-ready-search-open i' => 'padding: {{TOP}}{{UNIT}} {{RIGHT}}{{UNIT}} {{BOTTOM}}{{UNIT}} {{LEFT}}{{UNIT}};',

                ],
                'separator' => 'before',

            ]
        );

        $this->add_group_control(
            Group_Control_Typography::get_type(),
            [
                'name'     => '__search_menu_contentpo_i_icon_li_typho',
                'label'    => esc_html__('Typography', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-open i,{{WRAPPER}} .element-ready-search-open span',

            ]
        );


        $this->end_controls_section();
        $this->start_controls_section(
            'menu_search_close_button_section',
            [
                'label' => esc_html__('Close Button', 'element-ready-lite'),
                'tab'   => Controls_Manager::TAB_STYLE,
            ]
        );

        $this->add_control(
            '__search_menu_close_icon_bgcolor__transition',
            [
                'label'      => __('Transition', 'element-ready-lite'),
                'type'       => Controls_Manager::SLIDER,
                'size_units' => ['px'],
                'range'      => [
                    'px' => [
                        'min'  => 0.1,
                        'max'  => 3,
                        'step' => 0.1,
                    ],
                ],
                'default' => [
                    'unit' => 'px',
                    'size' => 0.3,
                ],
                'selectors' => [
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button' => 'transition: {{SIZE}}s;',
                ],
            ]
        );

        $this->add_control(
            '__search_menu_close_icon_bgcolor_',
            [

                'label'         => esc_html__('Close Button Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button span' => 'background: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button' => 'color: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            'slose__search_menu_i_icon_bgcolor_',
            [

                'label'         => esc_html__('Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button' => 'background: {{VALUE}};',

                ],
            ]
        );

        $this->add_group_control(
            \Elementor\Group_Control_Border::get_type(),
            [
                'name' => 'slose__search_menu_i_icon_border_color_',
                'selector' => '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button',
            ]
        );


        $this->add_control(
            '__search_box_menu_iqwert_icon_bgcolor_font_size',
            [
                'label' => esc_html__('Icon Font Size', 'element-ready-lite'),
                'type' => Controls_Manager::SLIDER,
                'size_units' => ['px', 'REM', 'em'],
                'range' => [
                    'px' => [
                        'min' => 0,
                        'max' => 150,
                        'step' => 5,
                    ],
                    '%' => [
                        'min' => 0,
                        'max' => 100,
                    ],
                ],

                'selectors' => [
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-form i' => 'font-size: {{SIZE}}{{UNIT}};',
                ],
            ]
        );

        $this->add_control(
            '_hover_bg_more_options',
            [
                'label' => esc_html__('Hover', 'element-ready-lite'),
                'type' => \Elementor\Controls_Manager::HEADING,
                'separator' => 'before',
            ]
        );

        $this->add_control(
            'slose__search_menu_i_icon_bgcolor_hover',
            [

                'label'         => esc_html__('Hover Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button:hover' => 'background: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_menu_close_icon_bgcolor_hover',
            [

                'label'         => esc_html__('Close Button Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button:hover span' => 'background: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button:hover' => 'color: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_menu_close_icon_boder_hover',
            [

                'label'         => esc_html__('Border Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-header .element-ready-search-close button:hover' => 'border-color: {{VALUE}};',

                ],
            ]
        );


        $this->end_controls_section();
        $this->start_controls_section(
            'menu_search_popup_inputs_section',
            [
                'label' => esc_html__('Input', 'element-ready-lite'),
                'tab'   => Controls_Manager::TAB_STYLE,

            ]
        );

        $this->add_control(
            '__search_box_menu_iqwert_icon_bgcolor_',
            [

                'label'         => esc_html__('Search icon Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-form i' => 'color: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-form svg path' => 'fill: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_box_menu_iqwert_input_color_',
            [

                'label'         => esc_html__('Input Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input' => 'color: {{VALUE}};',
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input::placeholder' => 'color: {{VALUE}};',

                ],
            ]
        );

        $this->add_control(
            '__search_box_menu_iqwert_input_color_foucs',
            [

                'label'         => esc_html__('Focus Color', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input:focus' => 'color: {{VALUE}};',
                ],
            ]
        );

        $this->add_group_control(
            Group_Control_Typography::get_type(),
            [
                'name'     => '__search_menu_conteo_i_input_typho',
                'label'    => esc_html__('Input Typography', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input,{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input::placeholder',

            ]
        );

        $this->add_control(
            '__search_box_menu_iqwert_input_bgcolor_',
            [

                'label'         => esc_html__('Input Background', 'element-ready-lite'),
                'type'         => Controls_Manager::COLOR,
                'selectors'     => [

                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input' => 'background: {{VALUE}};',

                ],
            ]
        );

        $this->add_responsive_control(
            '__search_box_menu_iqwert_input__border_radius',
            [
                'label'     => esc_html__('Border Radius', 'element-ready-lite'),
                'type'      => Controls_Manager::DIMENSIONS,
                'size_units' => ['px', '%', 'em'],
                'selectors' => [
                    '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input' => 'border-radius: {{TOP}}px {{RIGHT}}px {{BOTTOM}}px {{LEFT}}px;',
                ],
                'separator' => 'before',
            ]
        );

        $this->add_group_control(
            \Elementor\Group_Control_Border::get_type(),
            [
                'name' => '__search_box_menu_iqwert_input_section_border',
                'label' => esc_html__('Border', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input',
            ]
        );

        $this->add_group_control(
            \Elementor\Group_Control_Box_Shadow::get_type(),
            [
                'name' => 'menu_ite__search_box_iqwert_inputbox_shadow',
                'label' => esc_html__('Box Shadow', 'element-ready-lite'),
                'selector' => '{{WRAPPER}} .element-ready-search-box .element-ready-search-body .element-ready-search-form input',
            ]
        );

        $this->end_controls_section();
    } //Register control end

    protected function render()
    {

        $settings       = $this->get_settings();
        $widget_id    = 'element-ready-' . $this->get_id() . '-';

?>
        <!--====== Header START ======-->
        <?php if ($settings['_style'] == 'style1') : ?>
            <?php include('layout/search/style1.php'); ?>
        <?php endif; ?>
        <!--====== PART ENDS ======-->
<?php

    }

    protected function content_template()
    {
    }
}
