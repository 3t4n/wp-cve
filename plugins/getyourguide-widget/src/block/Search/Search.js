import React, { Component } from 'react';
import escape from 'lodash.escape';
import uniqueID from 'lodash.uniqueid';
import { classNames, spacers, gygData } from '../config';

export const addSearchStyle = () => {
  const id = 'gyg-search-style';
  if (document.getElementById(id)) {
    return;
  }
  document.head.insertAdjacentHTML(
    'beforeend',
    `<style id="${id}">@font-face {
        font-family: "GT Eesti";
        font-style: normal;
        font-display:fallback;
        font-weight: 400;
        src: url("https://cdn.getyourguide.com/static/49829f4a1f87/customer/desktop/cached/fonts/GT-Eesti/GT-Eesti-Pro-Display-Regular.woff2");
      }</style>`,
  );
};

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.formRef = React.createRef();
    this.id = `gyg-search-input-${uniqueID()}`;
  }

  state = {
    q: '',
  };

  componentDidMount() {
    addSearchStyle();
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.setState({ q: escape(e.target.value) });
    this.formRef.current.submit();
  };

  render() {
    const { formRef, handleSubmit, id } = this;
    const { partnerID: partner_id } = gygData; // eslint-disable-line camelcase
    const { q } = this.state;
    const didSubmit = q !== '';
    return (
      <form
        ref={formRef}
        action="https://www.getyourguide.com/s"
        className={classNames.searchForm}
        onSubmit={handleSubmit}
        style={{
          backgroundColor: '#1a2b49',
          borderRadius: '2px',
          fontFamily: '"GT Eesti",Arial,sans-serif',
          marginBottom: spacers.standard,
          padding: '24px',
        }}
      >
        <input
          className={classNames.searchQueryInput}
          type="hidden"
          name={classNames.searchQueryInput}
          value={q}
        />
        <input
          className={classNames.partnerId}
          type="hidden"
          name={classNames.partnerId}
          value={partner_id} // eslint-disable-line camelcase
        />
        <label htmlFor={id} style={{ flex: '1 0 auto' }}>
          <h2
            style={{
              color: '#fff',
              fontSize: '24px',
              margin: '0 0 12px',
              padding: 0,
            }}
          >
            Search GetYourGuide
          </h2>
        </label>
        <span style={{ display: 'flex', maxHeight: '48px' }}>
          <input
            id={id}
            placeholder="Where are you going?"
            style={{
              border: 0,
              borderRadius: '2px',
              fontSize: '16px',
              marginRight: '16px',
              padding: '8px',
              width: '100%',
            }}
            className={classNames.searchInput}
          />
          <button
            style={{
              appearance: 'none',
              backgroundColor: didSubmit ? '#0079e1' : '#1593ff',
              color: '#fff',
              border: 0,
              borderRadius: '5rem',
              fontSize: '16px',
              padding: 0,
              textTransform: 'none',
            }}
            type="submit"
          >
            <span style={{ display: 'flex', alignItems: 'center', padding: '8px' }}>
              {!didSubmit && (
                <svg
                  style={{
                    fill: '#fff',
                    marginRight: '8px',
                    maxHeight: '16px',
                    maxWidth: '16px',
                  }}
                  width="1792"
                  height="1792"
                  viewBox="0 0 1792 1792"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path d="M1216 832q0-185-131.5-316.5t-316.5-131.5-316.5 131.5-131.5 316.5 131.5 316.5 316.5 131.5 316.5-131.5 131.5-316.5zm512 832q0 52-38 90t-90 38q-54 0-90-38l-343-342q-179 124-399 124-143 0-273.5-55.5t-225-150-150-225-55.5-273.5 55.5-273.5 150-225 225-150 273.5-55.5 273.5 55.5 225 150 150 225 55.5 273.5q0 220-124 399l343 343q37 37 37 90z" />
                </svg>
              )}
              {didSubmit ? 'Loading...' : 'Search'}
            </span>
          </button>
        </span>
      </form>
    );
  }
}
