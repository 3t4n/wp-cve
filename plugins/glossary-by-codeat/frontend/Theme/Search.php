<?php

/**
 * Glossary
 *
 * @package   Glossary
 * @author    Codeat <support@codeat.co>
 * @copyright 2020
 * @license   GPL 3.0+
 * @link      https://codeat.co
 */

namespace Glossary\Frontend\Theme;

use Glossary\Engine;

/**
 * Extends the native search
 */
class Search extends Engine\Base {

	/**
	 * Initialize the class with all the hooks
	 *
	 * @since 1.0.0
	 * @return bool
	 */
	public function initialize() {
		parent::initialize();

		if ( !isset( $this->settings[ 'search' ] ) ) {
			return false;
		}

		\add_filter( 'pre_get_posts', array( $this, 'filter_search' ) );

		return true;
	}

	/**
	 * Add support for custom CPT on the search box
	 *
	 * @param object $query The query.
	 * @since 1.0.0
	 * @return object The query with changes.
	 */
	public function filter_search( $query ) {
		if ( $query->is_search && !\is_admin() ) {
			$post_types = $query->get( 'post_type' );

			if ( 'post' === $post_types ) {
				$post_types = array();
				$query->set( 'post_type', \array_push( $post_types, $this->default_parameters['post_type'] ) );
			}
		}

		return $query;
	}

}
