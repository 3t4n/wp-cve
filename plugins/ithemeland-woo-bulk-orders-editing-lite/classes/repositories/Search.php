<?php

namespace wobel\classes\repositories;

class Search
{
    private $filter_profile_option_name = "wobel_filter_profile";
    private $use_always_table = "wobel_filter_profile_use_always";
    private $current_data_option_name = "wobel_filter_profile_current_data";

    public function update(array $data)
    {
        if (!isset($data['key'])) {
            return false;
        }

        $presets = $this->get_presets();
        $presets[$data['key']] = $data;
        return update_option($this->filter_profile_option_name, $presets);
    }

    public function delete($preset_key)
    {
        $presets = $this->get_presets();
        if (is_array($presets) && array_key_exists($preset_key, $presets)) {
            unset($presets[$preset_key]);
        }
        return update_option($this->filter_profile_option_name, $presets);
    }

    public function get_preset($preset_key)
    {
        $presets = $this->get_presets();
        return (isset($presets[esc_sql($preset_key)])) ? $presets[esc_sql($preset_key)] : false;
    }

    public function get_presets()
    {
        return get_option($this->filter_profile_option_name);
    }

    public function update_use_always(string $preset_key, string $option_name = '')
    {
        $option_name = (!empty($option_name)) ? esc_sql($option_name) : $this->use_always_table;
        return update_option($option_name, esc_sql($preset_key));
    }

    public function get_use_always()
    {
        return get_option($this->use_always_table);
    }

    public function get_current_data()
    {
        return get_option($this->current_data_option_name);
    }

    public function update_current_data($current_data)
    {
        if (empty($current_data) || !is_array($current_data)) {
            return false;
        }

        $old_current_data = $this->get_current_data();
        if (empty($old_current_data) || !is_array($old_current_data)) {
            $old_current_data = [];
        }
        foreach ($current_data as $data_key => $data_value) {
            $old_current_data[esc_sql($data_key)] = esc_sql($data_value);
        }

        return update_option($this->current_data_option_name, $old_current_data);
    }

    public function delete_current_data()
    {
        return delete_option($this->current_data_option_name);
    }

    public function has_search_options()
    {
        $filters = get_option($this->filter_profile_option_name);
        $use_always = get_option($this->use_always_table);

        return (!empty($filters) && !empty($use_always));
    }

    public function set_default_item()
    {
        $default_item['default'] = [
            'name' => esc_html__('All Orders', 'ithemeland-woocommerce-bulk-orders-editing-lite'),
            'date_modified' => date('Y-m-d H:i:s', time()),
            'key' => 'default',
            'filter_data' => []
        ];
        $this->update_use_always('default');
        return update_option($this->filter_profile_option_name, $default_item);
    }
}
