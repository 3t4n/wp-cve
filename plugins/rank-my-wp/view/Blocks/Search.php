<div class="header-search">
    <div class="lsvr-container">
        <div class="header-search__inner">
            <form class="header-search-form header-search-form--ajax header-search-form--has-search-filter header-search-form--has-keywords header-search-form--has-rating header-search-form--rating-type-likes" id="header-search-form-1" action="https://howto.rankmywp.com/" method="get">
                <div class="header-search-form__inner">
                    <div class="header-search-form__content">
                        <div class="header-search-form__input-wrapper">
                            <input class="header-search-form__input" type="text" name="s" autocomplete="off" placeholder="Search the Knowledge Base" value="">
                            <button class="header-search-form__submit" type="button" title="Search">
                                <i class="fa fa-search" style="font-size: 20px;"></i></button>
                            <div class="header-search-form__spinner c-spinner"></div>
                        </div>
                        <div class="header-search-form__panel-inner">
                        </div>
                    </div>
                    <div class="header-search-form__keywords">
                        <div class="header-search-form__keywords-inner">
                            <span class="header-search-form__keywords-label">Suggested Search:</span>
                            <a href="https://howto.rankmywp.com/?s=keyword%20research" data-search-keyword="keyword research">keyword research</a>,
                            <a href="https://howto.rankmywp.com/?s=briefcase" data-search-keyword="briefcase">briefcase</a>,
                            <a href="https://howto.rankmywp.com/?s=ranking" data-search-keyword="ranking">ranking</a>
                        </div>
                    </div>
                </div>
            </form>
        </div>
    </div>
</div>