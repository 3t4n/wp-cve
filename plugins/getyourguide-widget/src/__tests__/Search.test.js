import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../block/Search/Search';

describe('Search', () => {
  it('renders correctly with no props', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
