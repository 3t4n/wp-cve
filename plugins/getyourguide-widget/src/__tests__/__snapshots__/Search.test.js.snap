// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly with no props 1`] = `
<form
  action="https://www.getyourguide.com/s"
  className="gyg-search"
  onSubmit={[Function]}
  style={
    Object {
      "backgroundColor": "#1a2b49",
      "borderRadius": "2px",
      "fontFamily": "\\"GT Eesti\\",Arial,sans-serif",
      "marginBottom": "1.5em",
      "padding": "24px",
    }
  }
>
  <input
    className="q"
    name="q"
    type="hidden"
    value=""
  />
  <label
    htmlFor="gyg-search-input-1"
    style={
      Object {
        "flex": "1 0 auto",
      }
    }
  >
    <h2
      style={
        Object {
          "color": "#fff",
          "fontSize": "24px",
          "margin": "0 0 12px",
          "padding": 0,
        }
      }
    >
      Search GetYourGuide
    </h2>
  </label>
  <span
    style={
      Object {
        "display": "flex",
        "maxHeight": "48px",
      }
    }
  >
    <input
      className="gyg-search-input"
      id="gyg-search-input-1"
      placeholder="Where are you going?"
      style={
        Object {
          "border": 0,
          "borderRadius": "2px",
          "fontSize": "16px",
          "marginRight": "16px",
          "padding": "8px",
          "width": "100%",
        }
      }
    />
    <button
      style={
        Object {
          "appearance": "none",
          "backgroundColor": "#1593ff",
          "border": 0,
          "borderRadius": "5rem",
          "color": "#fff",
          "fontSize": "16px",
          "padding": 0,
          "textTransform": "none",
        }
      }
      type="submit"
    >
      <span
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "padding": "8px",
          }
        }
      >
        <svg
          height="1792"
          style={
            Object {
              "fill": "#fff",
              "marginRight": "8px",
              "maxHeight": "16px",
              "maxWidth": "16px",
            }
          }
          viewBox="0 0 1792 1792"
          width="1792"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1216 832q0-185-131.5-316.5t-316.5-131.5-316.5 131.5-131.5 316.5 131.5 316.5 316.5 131.5 316.5-131.5 131.5-316.5zm512 832q0 52-38 90t-90 38q-54 0-90-38l-343-342q-179 124-399 124-143 0-273.5-55.5t-225-150-150-225-55.5-273.5 55.5-273.5 150-225 225-150 273.5-55.5 273.5 55.5 225 150 150 225 55.5 273.5q0 220-124 399l343 343q37 37 37 90z"
          />
        </svg>
        Search
      </span>
    </button>
  </span>
</form>
`;
